src/test/java/com/melon/portfoliomanager/services/AnalyticsServiceTest.java

```diff
@@ -0,0 +1,188 @@
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ExtendWith(OutputCaptureExtension.class)
+public class AnalyticsServiceTest {
+
+    @MockBean
+    private TransactionRepository transactionRepository;
+
+    @Mock
+    private PortfolioItemRepository portfolioItemRepository;
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private StockPricesMockApiHttpService stockPricesMockApiHttpService;
+
+    @InjectMocks
+    private AnalyticsService analyticsService;
+
+    @Test
+    public void fetchAnalyticsForUser_NonExistentUser_ThrowsException() {
+        when(userRepository.findByUsername("test1")).thenReturn(Collections.emptyList());
+
+        assertThrows(NoSuchUserException.class, () -> {
+            analyticsService.fetchAnalyticsForUser("test1");
+        });
+    }
+
+    @Test
+    public void fetchAnalyticsForUser_UserExistsButNoPortfolioItems_ReturnsZeroValues() throws Exception {
+        User user = new User();
+        user.setId(1L);
+        when(userRepository.findByUsername("testUser")).thenReturn(Collections.singletonList(user));
+        when(portfolioItemRepository.findByUserId(1L)).thenReturn(Collections.emptyList());
+
+        AnalyticsResponse response = analyticsService.fetchAnalyticsForUser("testUser");
+
+        assertNotNull(response);
+        assertEquals(0, response.getTotalPortfolioValue());
+        assertEquals(0, response.getTotalPortfolioGainVal());
+        assertEquals(0, response.getTotalPortfolioGainPct());
+        assertTrue(response.getAssets().isEmpty());
+    }
+
+
+    @Test
+    public void fetchAnalyticsForUser_ValidScenario_ReturnsCorrectValues() throws Exception {
+        User user = new User();
+        user.setId(1L);
+
+        PortfolioItem item = new PortfolioItem();
+        item.setCompanyName("Apple");
+        item.setQuantity(5.0);
+        item.setTotalBoughtPrice(400.0);
+        item.setTotalSoldPrice(0.0);
+
+        Map<String, Double> stockPrices = new HashMap<>();
+        stockPrices.put("Apple", 100.0);
+
+        StockPricesDTO stockPricesDTO = new StockPricesDTO();
+        stockPricesDTO.setStockPrices(stockPrices);
+
+        ResponseEntity<StockPricesDTO> wrappedDto = new ResponseEntity<>(stockPricesDTO, HttpStatusCode.valueOf(200));
+
+        when(userRepository.findByUsername("testUser")).thenReturn(Collections.singletonList(user));
+        when(portfolioItemRepository.findByUserId(1L)).thenReturn(Collections.singletonList(item));
+        when(stockPricesMockApiHttpService.getStockPrices()).thenReturn(wrappedDto);
+
+        AnalyticsResponse response = analyticsService.fetchAnalyticsForUser("testUser");
+
+        assertNotNull(response);
+        assertEquals(500, response.getTotalPortfolioValue());
+        assertEquals(100, response.getTotalPortfolioGainVal());
+
+    }
+
+    @Test
+    public void fetchAnalyticsForUser_MultipleAssets_ReturnsCorrectValues() throws Exception {
```

Comment:
This test is pretty huge. Please, try to extract some parts of it in separate methods so that is becomes a bit more readable.

Reply:
fixed